src/dataFiles.ts

```diff
@@ -1,16 +1,30 @@
 import { readdirSync, statSync } from "fs";
-import { extname, join } from "path";
-import { DATA_EXTENSIONS } from "./constants.js";
+import { basename, extname, join } from "path";
+import {
+	BADFOLDERS_REGEX,
+	IGNORED_FOLDERS,
+	VIDEO_EXTENSIONS,
+} from "./constants.js";
 import { getRuntimeConfig } from "./runtimeConfig.js";
 
+function checkIgnorePath(root: string, isDir: boolean) {
+	return (
+		(isDir &&
+			(IGNORED_FOLDERS.includes(basename(root).toLowerCase()) ||
+				BADFOLDERS_REGEX.test(basename(root)))) ||
+		(!isDir && !VIDEO_EXTENSIONS.includes(extname(basename(root))))
```

Comment:
can we refactor this `isDir && foo || !isDir && bar` into an `if` for readability? 

```js
const name = basename(root);
if (isDir) {
  return BADFOLDERS_REGEX.test(name) || IGNORED_FOLDERS.includes(name.toLowerCase())
} else {
  return !VIDEO_EXTENSIONS.includes(extname(basename(root))
}
```